### DISCUSSION TITLE
Allow usage of S3 temporary files without S3 upload links

### DISCUSSION THREAD

**Discussion by apreiml** [livewire/livewire#6778] [3. Feature ideas]
If S3 is enabled as disk driver, temporary file uploads uploads will go directly to S3 via generated S3 links.

Sometimes it's desired to use S3, but still let temporary files be uploaded using the Livewire Controllers. For example if someone hosts a custom S3 solution that is hidden behind a firewall or when CSP rules are in place, that do not allow the S3 urls.

Hence I propose to add an option to the temporary_file_upload that disables direct s3 links. I've submitted a PR for V2 that does this: https://github.com/livewire/livewire/pull/6513

If the option is not accepted, applying only commit https://github.com/livewire/livewire/pull/6513/commits/e35a638552db42cd1e6d935439f669eead8d0555 will allow to use filesystems other than s3 that do not store files locally. A workaround would be then to register the S3 driver under another name locally in the user projects to bypass s3 direct URLs.

**Comment by apreiml** [comment]
https://github.com/livewire/livewire/discussions/7466

**Comment by emkcloud** [comment]
I also needed to store temporary files on S3 since I'm behind a load balancer, and the server executing the save operation might not be the same as the one storing the temporary file. Therefore, I need to use S3 as if it were a local filesystem, without the browser performing the upload operation.

**composer require league/flysystem-path-prefixing**

For now, I’ve found this solution: modify the file config/filesystems.php and add a disk.

```php
'disks' =>
[
    // LIST YOUR DISKS ...
    // LIST YOUR DISKS ...

   'temporary'      =>
   [
         'driver'     => 'scoped',
         'disk'       => 'your-disk-s3', // Name of disk with driver s3
         'prefix'     => 'temporary',
    ],
]
```
Change the config/livewire.php

```php
'temporary_file_upload' =>
[
    'disk'       => 'temporary',
    'directory'  => 'livewire',
    'rules'      => 'file|mimes:png,jpg,pdf|max:102400',
    'middleware' => 'throttle:10,1',
]
```

**Test and results with S3 bucket**

![image](https://github.com/user-attachments/assets/193c956f-ca5c-467e-be90-6e6581c1d7eb)

**Reply by plorenz-etes** [reply]
Thank you, you saved my day with this!
